app/views/projects/blob/new.html.haml

```diff
@@ -1,5 +1,11 @@
-- page_title "New File", @ref
-%h3.page-title New file
+%h3.page-title<
+  Create new file or 
+  %a.button{:href=>'#modal-upload-blob', 'data-target'=>'#modal-upload-blob', 'data-toggle'=>'modal'}> &nbspupload existing one
```

Comment:
`href` should use Ruby 1.9 Hash syntax.

You're also missing the semicolon for the `&nbsp;` entity.